lookout/core/metrics/server.py

```diff
@@ -0,0 +1,225 @@
+        self.default_metric = default_metric
+
+    @property
+    def registry(self):
+        return self.__registry
+
+    @property
+    def port(self) -> int:
+        return self.__port
+
+    @property
+    def addr(self) -> str:
+        return self.__addr
+
+    @property
+    def metrics(self) -> dict:
+        return self.__metrics
+
+    @property
+    def is_running(self) -> bool:
+        return self.__is_running
+
+    def start_http_server(self):
+        start_http_server(port=self.port, addr=self.addr, registry=self.registry)
+        self.__is_running = True
+
+    def create_new_metric(
+        self, name: str, labelnames: str = "", metric_type=None, *args, **kwargs
+    ):
+        """Create a new metric in case it does not preivously exists.
+
+        :param name: It will be used as a key to identify the new metric.
+        :param labelnames: Additional description of the event. Only used when creating a new
+            event.
+        :param metric_type: Type of metric that will be used to register the target value. Only
+            used when creating a new event.
+        :param args: Additional args to initialize the target metric_type.
+        :param kwargs: Additional kwargs to initialize the target metric_type.
+        :return: None
+        """
+        metric_type = self.default_metric if metric_type is None else metric_type
+        self.metrics[name] = metric_type(name, labelnames, registry=self.registry, *args, **kwargs)
+
+    def submit_event(
+        self,
+        key: str,
+        value: Union[int, float, bool],
+        update_method: str = None,
+        metric_type: Callable = None,
+        *args,
+        **kwargs,
+    ):
+        """Register an event by a key and with a numeric value. If the key does not exist, it
+        creates a new Prometheus Metric.
+
+        :param key: Identifier of the event.
+        :param value: Value of the event. It will convert cast variables to int.
+        :param metric_type: Type of metric that will be used to register the target value. Only
+            used when creating a new event.
+        :param update_method: Function to be called on the target metric to update the values.
+            If None it will try to apply update the metric calling observe(), inc() if
+            observe is not available, and set() if inc is not available.
+        :param args: Additional args to initialize the target metric_type. Ignored if no new metric
+            is created.
+        :param kwargs: Additional kwargs to initialize the target metric_type. Ignored if no new
+            metric is created.
+        :return: None
+        """
+        if key not in self.metrics.keys():
+            self.create_new_metric(name=key, metric_type=metric_type, *args, **kwargs)
+        value = int(value) if isinstance(value, bool) else value
+        if hasattr(self.metrics[key], "observe") and update_method is None:
+            self.metrics[key].observe(value)
```

Comment:
All of these feels too complex, remove everything. No set. No defaults. Always `add`.

Reply:
This implies loosing compatibility with the standard API of the prometheus_client.

 Other side effects are: restricting the usage to only our custom metric, and only being able to use our custom metric with the method add.

Reply:
Totally fine